include/reader.h

```diff
@@ -37,6 +37,38 @@ using namespace std;
 namespace kiwix
 {
 
+/**
+ * The SuggestionItem is a helper class that contains the info about a single
+ * suggestion item.
+ */
+
+class SuggestionItem
+{
+  // Functions
+  public:
+    // Create a sugggestion item.
+    explicit SuggestionItem(std::string title, std::string normalizedTitle,
+                            std::string path, std::string snippet = "") :
+      title(title),
+      normalizedTitle(normalizedTitle),
+      path(path),
+      snippet(snippet) {}
```

Comment:
This is useless for the user to create a suggestion itself.
It is better to make the constructor private and add the `Reader` as friend class.